server/routes/creatingReports/reportMayAlreadyExist.ts

```diff
@@ -0,0 +1,73 @@
+import { Request, Response, RequestHandler } from 'express'
+import moment from 'moment'
+import type LocationService from '../../services/locationService'
+import type ReportService from '../../services/reportService'
+import type OffenderService from '../../services/offenderService'
+import { SystemToken } from '../../types/uof'
+import logger from '../../../log'
+
+export default class ReportMayAlreadyExistRoutes {
+  constructor(
+    private readonly systemToken: SystemToken,
+    private readonly reportService: ReportService,
+    private readonly locationService: LocationService,
+    private readonly offenderService: OffenderService
+  ) {}
+
+  public view = async (req: Request, res: Response): Promise<void> => {
+    const { bookingId } = req.params
+    const token = await this.systemToken(res.locals.user.username)
+    const { displayName: offenderName } = await this.offenderService.getOffenderDetails(token, parseInt(bookingId, 10))
+    const inProgressReport = await this.reportService.getReportInProgress(parseInt(bookingId, 10))
+
+    if (!inProgressReport) {
+      logger.info('User has attempted to access IN_PROGRESS report which is no longer available')
```

Comment:
I don't really understand this logic and not sure the log message is helping me.

Do you mean any IN_PROGRESS report for this offender or the user's current report?   

If you want to retrieve the current user's report - you need to search for it using the compound key of `user_Id`, 'booking_number' and latest sequence id - I think there's something like `getCurrentDraft` for this.

Reply:
The reason for doing this was to prevent user navigating back from report-cancelled to report-may-already-exist pages. Hence i need to get this user's current rather than 'any', in_progress report.